### ISSUE TITLE
Add wrappers for aio functions

### ISSUE THREAD

**Issue by adammoody** [LLNL/UnifyCR#225]
We have a wrapper for lio_listio(), but its implementation does not handle all I/O ops (read ops, but not write ops), nor does it support the user providing a list of ops that might mix across unifycr and system files.  Additionally, we likely need wrappers for other aio functions, and we should consider how to make the implementations actually be asynchronous, e.g., start the read/write op but not block on it to complete.

In short, we need to spend some time considering aio wrappers.  Or we could opt to decide to not support async I/O operations, in which case, we should probably drop our lio_listio wrapper.

**Comment by MichaelBrim** [comment]
The 'lio_listio()' wrapper now supports both read and write, with a mixture of regular and UnifyCR files. We still don't support asynchronous mode (i.e., `LIO_NOWAIT`) though.
